aws/provider_test.go

```diff
@@ -1036,6 +1036,66 @@ func testAccErrorCheckSkipMessagesContaining(t *testing.T, messages ...string) r
 	}
 }
 
+type ServiceErrorCheckFunc func(*testing.T) resource.ErrorCheckFunc
+
+var serviceErrorCheckFuncs map[string]ServiceErrorCheckFunc
+
+func RegisterServiceErrorCheckFunc(endpointID string, f ServiceErrorCheckFunc) {
+	if serviceErrorCheckFuncs == nil {
+		serviceErrorCheckFuncs = make(map[string]ServiceErrorCheckFunc)
+	}
+	serviceErrorCheckFuncs[endpointID] = f
```

Comment:
We may want to consider `panic()` for cases if a duplicate service is being registered to prevent that type of coding issue instead of potentially confusing behavior. I believe `init()` functions run in file naming alphabetical order.

Reply:
Good idea! That would be very confusing to figure out why a registered-over func wasn't running.